zanata-war/src/main/java/org/zanata/rest/RestLimitingSynchronousDispatcher.java

```diff
@@ -45,18 +71,33 @@ public RestLimitingSynchronousDispatcher(
         this.processor = processor;
     }
 
+    HttpServletRequest getServletRequest() {
+        return ServletContexts.instance().getRequest();
+    }
+
     @Override
     public void invoke(final HttpRequest request, final HttpResponse response) {
 
+        //This is for REST request from browser where user is logged in.
         HAccount authenticatedUser = getAuthenticatedUser();
-        String apiKey = HeaderHelper.getApiKey(request);
+
+        String apiKey = HttpUtil.getApiKey(request);
 
         try {
-            // we are not validating api key but will rate limit any api key
-            if (authenticatedUser == null && Strings.isNullOrEmpty(apiKey)) {
-                response.sendError(
-                        Response.Status.UNAUTHORIZED.getStatusCode(),
-                        API_KEY_ABSENCE_WARNING);
+            if (authenticatedUser == null
+                    && Strings.isNullOrEmpty(apiKey)
+                    && !SecurityFunctions.canAccessRestPath(authenticatedUser,
+                            request.getHttpMethod(),
+                            request.getPreprocessedPath())) {
```

Comment:
Every incoming request should trigger a call to `SecurityFunctions.canAccessRestPath`, not just the anonymous ones.